gcc/rust/ast/rust-ast-fragment.h

```diff
@@ -117,6 +122,14 @@ class Fragment
   bool is_single_fragment_of_kind (SingleASTNode::NodeType expected) const;
   void assert_single_fragment (SingleASTNode::NodeType expected) const;
 };
+
+/**
+ * This is the type for transcriber functions found in
+ * rust-macro-builtins.{h,cc}.
+ */
+typedef std::function<tl::optional<Fragment> (Location, MacroInvocData &)>
+  transcriber_t;
```

Comment:
I think the C++ way to do this is
```suggestion
using transcriber_t = std::function<tl::optional<Fragment> (Location, MacroInvocData &)>;
```

I think it would also be nice to have that `typedef`/`using` be part of a class, like `MacroRulesDefinition`. This way, the type would be even more namespaced, and we could use it by writing `MacroRulesDefinition::Transcriber` or `MacroRulesDefinition::transcriber_t`. I'm saying this mostly because of...

Reply:
done